pkg/controller/v1/xgboostjob/pod.go

```diff
@@ -110,6 +126,16 @@ func SetPodEnv(job interface{}, podTemplate *corev1.PodTemplateSpec, rtype, inde
 			Name:  "PYTHONUNBUFFERED",
 			Value: "0",
 		})
+		if totalReplicas > 1 {
```

Comment:
Add comment clarifying that these env vars are necessary for LightGBM only.

Reply:
[Added](https://github.com/kubeflow/xgboost-operator/pull/103/commits/d90f9452746cec231de25352029094043b127c4b)